org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/DebugUtility.java

```diff
@@ -136,6 +138,22 @@ public static IDebugSession attach(VirtualMachineManager vmManager, String hostN
     public static CompletableFuture<Location> stepOut(ThreadReference thread, IEventHub eventHub) {
         return DebugUtility.step(thread, eventHub, StepRequest.STEP_LINE, StepRequest.STEP_OUT);
     }
+    
+    /**
+     * Send exception request.
+     * @param manager
+     *               the event request manager
+     * @param notifyCaught
+     *               whether to notify caught exceptions
+     * @param notifyUncaught
+     *               whether to notify uncaught exceptions
+     */
+    public static void sendExceptionRequest(EventRequestManager manager, boolean notifyCaught, boolean notifyUncaught) {
+        List<ExceptionRequest> legacy = manager.exceptionRequests();
+        ExceptionRequest request = manager.createExceptionRequest(null, notifyCaught, notifyUncaught);
+        request.enable();
+        manager.deleteEventRequests(legacy);
```

Comment:
No. Your code maybe have potential bug. You know the legacy is a list and it is a reference. And the newly created request may be in the legacy list. Put delete in the last step, it could delete new request too. 
Besides, what kind of exception could be thrown for create operation?